container_test.go

```diff
@@ -127,6 +128,50 @@ func Test_GetDockerfile(t *testing.T) {
 	}
 }
 
+func Test_GetAuthConfigs(t *testing.T) {
```

Comment:
I see the value in this test to verify that the auth credentials are set, and I'd like to see a test for interacting with a real registry. 

Have you explored the idea of starting a Docker registry as a container, configured with security as explained here https://docs.docker.com/registry/deploying/#restricting-access?

Reply:
Yes. I just tried it and updated this PR. Could you please take a look?